src/presentation/pages/NoteSettings.vue

```diff
@@ -64,23 +56,21 @@ const invitationLink = computed(
 loadSettings(props.id);
 
 /**
- * Button click handler
+ * Regenerate invitation hash
  */
-function onClick() {
-  if (!noteSettings.value) {
-    throw new Error('Note settings is not loaded');
-  }
-  updateSettings(props.id, {
-    isPublic: noteSettings.value.isPublic,
-    customHostname: noteSettings.value.customHostname,
-  });
+async function regenerateHash() {
+  revokeHash(props.id);
 }
 
 /**
- * Regenerate invitation hash
+ * Change isPublic property
  */
-async function regenerateHash() {
-  revokeHash(props.id);
+async function changeAccess() {
+  const isPublic = noteSettings.value?.isPublic;
+
+  if (isPublic !== undefined) {
```

Comment:
if user changes access, it means that note settings is defined. So this case has no sense

Reply:
But we still should to check noteSettongs.value, in other case we have error

Reply:
I'd suggest using `!` instead, since we know that it can't be undefined here.

Reply:
Okay